File: python/anagram/anagram.py
def detect_anagrams(word, candidates):
    result = [match(w, word) for w in candidates]
    return list(filter(None, result))

def match(list_word, input_word):
    if list_word.lower() != input_word.lower() and letters_of(list_word) == letters_of(input_word):
        return list_word

def letters_of(word):
    l = list(word.lower())
    l.sort()
    return l

File: ruby/anagram/anagram.rb
class Anagram
  def initialize(word)
    @input_word = word
  end

  def match(match_list)
    match_list.select do |w|
      w.downcase != @input_word.downcase &&
        letters_of(w) == letters_of(@input_word)
    end
  end

  private

  def letters_of(word)
    word.downcase.each_char.sort
  end
end

module BookKeeping
  VERSION = 2
end
